send.go

```diff
@@ -115,8 +117,14 @@ func (c *Connection) send(ctx context.Context, request *http.Request) (response
 		return
 	}
 
+	var token string
 	// Get the access token:
-	token, _, err := c.TokensContext(ctx)
+	operation := func() error {
+		token, _, err = c.TokensContext(ctx)
+		return err
+	}
```

Comment:
We should retry this only in situations where we know that retry may help. It will not help, for example, if the credentials are incorrect. Can we limit the retry only to situations where the SSO service returns certain error responses?

I'd also suggest to move the retry logic inside the `TokensContext` method, so that users that call directly that method also benefit from that logic.

Reply:
@jhernand do you know what error responses should trigger a retry?

For the second suggestion - Fixed. Thanks